pkg/processor/runtime/rpc/connection/connectionallocator.go

```diff
@@ -132,17 +105,97 @@ func (ca *ConnectionAllocator) Stop() error {
 }
 
 func (ca *ConnectionAllocator) Allocate(duration time.Duration) (eventprocessor.EventProcessor, error) {
-	// TODO: support multiple connections
 	return ca.allocator.Allocate(duration)
 }
 
+func (ca *ConnectionAllocator) Release(processor eventprocessor.EventProcessor) {
+	// if when releasing processor requires restart, recreate the connection
+	if processor.GetStatus() == status.RestartRequired {
+		var newProcessor []eventprocessor.EventProcessor
+		var err error
+
+		// Retry connection creation up to 3 times
+		for i := 0; i < 3; i++ {
+			newProcessor, err = ca.createConnections(1)
+			if err == nil {
+				break
+			}
+			ca.Logger.WarnWith("Failed to recreate connection, retrying", "attempt", i+1, "error", err)
+		}
+
+		// If still failing after retries, log the error and set status to not ready (it will signal to wrapper that restart is needed)
+		if err != nil {
+			ca.Logger.WarnWith("Failed to recreate connection after retries", "error", err)
+
+			// TODO: add a background check which checks connection manager status and restarts wrapper if needed
+			// it's only added to timeout.go which checks for event timeouts
+			// for now it's fine, however if we want to add more cases when restart is needed, then it should be done separately
+			ca.SetStatus(status.RestartRequired)
+		} else {
+			processor = newProcessor[0]
+		}
+	}
+	ca.allocator.Release(processor)
+}
+
 func (ca *ConnectionAllocator) GetAddressesForWrapperStart() ([]string, string) {
 	controlAddress := ""
 	if ca.controlMessageSocket != nil {
 		controlAddress = ca.controlMessageSocket.Address
 	}
 	return []string{ca.serverAddress}, controlAddress
 }
+
+// ConnectionMonitor checks that all connections in allocator are healthy
+// if some are not, it will re-establish them
+func (ca *ConnectionAllocator) ConnectionMonitor() {
+	// by default, wait for 5 seconds before the next check
+	timeout := 5 * time.Second
+	if ca.GetConfig().eventTimeout != 0 {
+		// if event timeout is set, then use it (because we want to find out about termination of a socket as soon as possible)
+		timeout = ca.GetConfig().eventTimeout
+	}
+	for {
+		ca.Logger.DebugWith("ConnectionsMonitor", "numObjectsAvailable", ca.allocator.GetNumObjectsAvailable())
+		time.Sleep(timeout)
+
+	}
```

Comment:
So... this does nothing? just log and sleep?

Reply:
This should be removed